### PR TITLE
Connorwon 214 resource endpoints

### PR DESCRIPTION
**Pull request by ConnorWon**
## Description

Created resource backend endpoints and implemented their corresponding functionality

### Changes

* Created schema for resource object
* Added CRUD endpoints for the new resource object
* Wrote tests for each of the newly implemented controller functions
* Changed 'hasMemberPermissions' in jest tests to 'hasFrontendAPIKey'
* Validate fn now makes changes to req.body if the validation schema provides default values for req.body fields

### Updates

[Please enter a brief summary of updates to this PR during code review - if any]

## Checklist

- [x] Ran `npm run lint:fix` to format code
- [x] Assigned PR to all authors
- [x] Requested at least one reviewer

### CODE REVIEW

**Comment by ConnorWon** [comment]
[add resource backend endpoint and functionality](https://trello.com/c/BY2em13c/214-add-resource-backend-endpoint-and-functionality)

File: controllers/resourceController.js [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+ *        }
+ * @param Responds with status code and messsage.
+ */
+export const createResource = async (req, res) => {
+	try {
+		const isMember = await hasFrontendAPIKey(req.headers.authorization);
+
+		if (!isMember) {
+			res.status(400).json({
+				message: 'Invalid access - must be a member to create a new resource.',
```
Comment (ElisaHu1) [inline controllers/resourceController.js:36]: what credential you use to access the website? 
it print out this error when I try CONTENT PLATFORM ROOT ADMIN USER

Comment (ConnorWon) [reply controllers/resourceController.js:36]: Yeah it produced that error for me as well. So what I did to test my code was I got rid of the hash checking in the hasFrontendAPIKey function in authHelper.js and just made the function return true as long as an access_token was provided

**Review by ryanyae — APPROVED** [review]
Great job on this ticket as well, what there is left is adding the other functionalities, thank you for laying the ground work!

**Pull request merged**